src/platform/backends/hyperv_api/virtdisk/virtdisk_wrapper.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include <hyperv_api/hyperv_api_operation_result.h>
+#include <hyperv_api/virtdisk/virtdisk_create_virtual_disk_params.h>
+#include <hyperv_api/virtdisk/virtdisk_disk_info.h>
+
+#include <multipass/singleton.h>
+
+namespace multipass::hyperv::virtdisk
+{
+
+/**
+ * A high-level wrapper class that defines
+ * the common operations that VirtDisk API
+ * provide.
+ */
+struct VirtDiskWrapper : public Singleton<VirtDiskWrapper>
+{
+    /**
+     * Construct a new VirtDiskWrapper
+     */
+    VirtDiskWrapper(const Singleton<VirtDiskWrapper>::PrivatePass&) noexcept;
+
+    // ---------------------------------------------------------
+
+    /**
+     * Create a new Virtual Disk
+     *
+     * @param [in] params Parameters for the new virtual disk
+     *
+     * @return An object that evaluates to true on success, false otherwise.
+     * message() may contain details of failure when result is false.
+     */
+    [[nodiscard]] virtual OperationResult create_virtual_disk(
+        const CreateVirtualDiskParameters& params) const;
+
+    // ---------------------------------------------------------
+
+    /**
+     * Resize an existing Virtual Disk
+     *
+     * @param [in] vhdx_path Path to the virtual disk
+     * @param [in] new_size New disk size, in bytes
+     *
+     * @return An object that evaluates to true on success, false otherwise.
+     * message() may contain details of failure when result is false.
+     */
+    [[nodiscard]] virtual OperationResult
+    resize_virtual_disk(const std::filesystem::path& vhdx_path, std::uint64_t new_size_bytes) const;
+
+    // ---------------------------------------------------------
+
+    /**
+     * Merge a child differencing disk to its parent
+     *
+     * @param [in] child Path to the differencing disk
+     *
+     * @return An object that evaluates to true on success, false otherwise.
+     * message() may contain details of failure when result is false.
+     */
+    [[nodiscard]] virtual OperationResult merge_virtual_disk_to_parent(
+        const std::filesystem::path& child) const;
```

Comment:
In the end, where is the result of the merger? I imagine it is in the child path, but the name seems to imply the direction is into the parent? I wouldn't mind seeing it spelled out in the doc and the name adjusted if appropriate (e.g. `merge_parent_into(child_disk)`).